include/alpaka/atomic/AtomicAtomicRef.hpp

```diff
@@ -0,0 +1,239 @@
+/* Copyright 2021 Felice Pantaleo (CERN), Andrea Bocci (CERN)
+ *
+ * This file is part of alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/atomic/Traits.hpp>
+#include <alpaka/core/BoostPredef.hpp>
+
+#include <boost/atomic.hpp>
+#include <type_traits>
+#include <array>
+
+namespace alpaka
+{
+    //! The CPU threads accelerator atomic ops.
+    //
+    //  Atomics can be used in the grids, blocks and threads hierarchy levels.
+    //  Atomics are not guaranteed to be save between devices.
```

Comment:
```suggestion
    //  Atomics are not guaranteed to be safe between devices.
```
?
It's not clear what "across multiple devices mean" for the `DevCpu` device, since there exists always only one, but these should be safe also across different cores and different sockets.

Reply:
thanks, of course this is safe across cores/sockets